netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -182,7 +183,13 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata.Headers he
     Http2Headers http2Headers = Utils.convertClientHeaders(headers, ssl, defaultPath, authority);
 
     // Write the command requesting the creation of the stream.
-    channel.writeAndFlush(new CreateStreamCommand(http2Headers, stream));
+    CreateStreamCommand msg = new CreateStreamCommand(http2Headers, stream);
+    if (method.getType().clientSendsOneMessage()) {
+      // No need to flush yet if the client is required to send at least one message
+      channel.write(msg);
```

Comment:
It wouldn't be too invasive, but it isn't as simple as adding one condition (which is the heart of this PR). I don't see any reason to make it part of this PR, and it would easily build off this PR.